src/cogent3/core/alphabet.py

```diff
@@ -56,38 +57,21 @@ class AlphabetError(Exception):
 
 
 def get_array_type(num_elements):
-    """Returns smallest array type that can contain sequence on num_elements.
-
-    Used to figure out how large a data type is needed for the array in which
-    elements are indices from an alphabet. If the data type is too small
-    (e.g. you allocated an uint8 array, with 256 possible states (0-255), but
-    your data actually have more than 256 states, e.g. tripeptide data with
-    20*20*20 = 8000 states), when you assign a state larger than the data type
-    can hold you'll get an unexpected result. For example, assigning state
-    800 in an array that can only hold 256 different states will actually
-    give you the result mod 256:
-
-    >>> a = array(range(10), uint8)
-    >>> a
-    array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9],'B')
-    >>> a[0] = 800
-    >>> a
-    array([32,  1,  2,  3,  4,  5,  6,  7,  8,  9],'B')
-           ^^
-           NOTE: element 1 is _not_ 800, but instead 32 -- nasty surprise!
-
-    Getting the size of the necessary array from the alphabet is a good
-    solution to this problem.
-
-    WARNING: Will not overflow if somehow you manage to feed it an alphabet
-    with more than 2**32 elements, but it seems unlikely that this will
-    happen very often in practice...
+    """Returns the smallest numpy integer dtype that can contain elements
+    within num_elements.
     """
     if num_elements <= 256:
-        return uint8
+        dtype = uint8
     elif num_elements <= 2 ** 16:
-        return uint16
-    return uint32
+        dtype = uint16
+    elif num_elements <= 2 ** 32:
+        dtype = uint32
```

Comment:
I think it should be
<= 255
<= 2**16 -1
<= 2**32 -1

or 

< 256
< 2**16
< 2**32

Reply:
Thanks for the review! You're absolutely correct!

Reply:
@GavinHuttley 
I wrote 2 ** 16 , but markup changed it to **2**16.